examples/user_guide/Lazy_Data.ipynb

```diff
@@ -0,0 +1,107 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notes (to be removed):\n",
+    "\n",
+    "- What is lazy data?\n",
+    "- Using dask with HoloViews\n",
+    "- Difference between .persist and .compute\n",
+    "  - `persist`: Caching intermediate results but allow further lazy evaluation\n",
+    "  - `compute`: Materialize the data into memory, e.g. ibis->pandas or dask->pandas\n",
+    "- When do you have to consider using persist?\n",
+    "  - Call persist at the end after aggregating or reducing your data in some way\n",
+    "  - Why?\n",
+    "    - Avoids plotting code from evaluating the graph multiple times\n",
+    "- When to consider using compute?\n",
+    "  - Certain interfaces don’t support some methods, e.g. `.sort`, `.iloc`, you might want to perform some operation and then materialize the data into a format that does support these operations\n",
+    "  \n",
+    "Question:  Need examples?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Lazy Data\n",
+    "\n",
+    "Lazy data, or lazy loading of data, is a technique used in which data is not actually loaded until it is needed.  If used effectively, this allows for a more efficient use of system resources and performance.  Additionally, a user can build up a list of tasks, or build a task graph, that uses the data, but doesn't actually run anything until told to do so.  The *expression* is stored instead of the result.\n",
```

Comment:
Maybe emphasize that lazy evaluation lets you separate describing what computations need to be done from actually doing them, and then have a bulleted or numbered list of why capturing such a pipeline lazily can be important:
- Lets you select different compute engines at the point the computation is done (e.g. to parallelize the computation, run it on GPUs, run it on a remote server, etc.)
- Lets you describe a series of operations where some of them might drastically reduce the size of the problem _before_ computation begins, e.g. by sampling/selecting/slicing a large dataset for a case when only a small portion is actually used
- Lets you run the operation "out of core", i.e., processing only a chunk at a time rather than loading all of the data at once, when the same operation must be done on many datapoints that together would not fit into memory

etc.